Atomic/syscontainers.py

```diff
@@ -2017,18 +2017,18 @@ def get_default_system_name(image):
 
     @staticmethod
     def _parse_imagename(imagename):
-        sep = imagename.find("/")
-        reg, image = imagename[:sep], imagename[sep + 1:]
-        if '.' not in reg:
-            # if the registry doesn't look like a domain, consider it as the
-            # image prefix
-            reg = ""
-            image = imagename
-        sep = image.find(":")
-        if sep > 0:
-            return reg, image[:sep], image[sep + 1:]
+        prefix, _, image = imagename.rpartition("/")
+        if "." in prefix:
+            # prefix looks like a domain, consider it a registry
+            reg, _, prefix = prefix.partition("/")
         else:
-            return reg, image, "latest"
+            reg = ""
+        image, _, tag = image.partition(":")
+        if prefix:
+            image = "%s/%s" % (prefix, image)
+        if not tag:
+            tag = "latest"
+        return reg, image, tag
```

Comment:
This was some old code that we forgot to refactor to `util.Decompose` when it was  introduced.  Do you think we can just use it here as well or does it do something different ?  Do your tests still pass?

Reply:
I updated `_parse_imagename` to use `util.Decompose`. Maybe it's worth to remove `_parse_imagename` altogether and add a getter to Decompose returning a repo with an image?

Reply:
I think there might be some unexpected effects because of `util.Decompose` setting `repo` to `library` if it's empty for docker.io images.